src/metabase/search/appdb/index.clj

```diff
@@ -265,14 +257,19 @@
         active-updated?  (safe-batch-upsert! (active-table) entries)
         pending-updated? (safe-batch-upsert! (pending-table) entries)]
     (when (or active-updated? pending-updated?)
-      (->> entries (map :model) frequencies))))
+      (u/prog1 (->> entries (map :model) frequencies)
+        (log/trace "indexed documents" <>)))))
 
 (defmethod search.engine/consume! :search.engine/appdb [_engine document-reducible]
   (transduce (comp (partition-all insert-batch-size)
                    (map batch-update!))
              (partial merge-with +)
              document-reducible))
 
+(defmethod search.engine/delete! :search.engine/appdb [_engine search-model ids]
```

Comment:
In your usage of this method you're passing the toucan model, not the search model. I think it's best to keep it that way, but do the look-up of the corresponding search-models here, so that it works 😄

Reply:
Ah nevermind, I see that's done between `core/delete!` and `engine/delete!`, a bit tricksy.

Reply:
yeah but I don't want to force every engine to implement this lookup, right?